### PR TITLE
Re-add dependency for older junit tests

### PR DESCRIPTION
**Pull request by matozoid**
The build missed a lot of tests for a few months. Let's see how big the damage is.

### CODE REVIEW

File: javaparser-core-testing/src/test/java/com/github/javaparser/ast/DataKeyTest.java [hunk 1]
```diff
@@ -47,7 +48,16 @@ public void addAFewKeysAndSeeIfTheyAreStoredCorrectly() {
...
+        assertThat(node.containsData(ABC)).isTrue();
+        assertThat(node.containsData(LISTY)).isTrue();
+        assertThat(node.containsData(DING)).isFalse();
+    }
+
+    @Test
+    void aNonExistentKeyThrowsAnException() {
+        Node node = new SimpleName();
+
+        assertThrows(IllegalStateException.class, () -> node.getData(DING));
```
Comment (matozoid) [inline javaparser-core-testing/src/test/java/com/github/javaparser/ast/DataKeyTest.java:60]: This is from my change to be able to see if something exists for a certain key.

File: javaparser-core-testing/src/test/java/com/github/javaparser/modules/ModuleDeclarationTest.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public void testPrettyPrinting() {
                         "  requires A.B;" +
                         "  requires transitive C.D;" +
                         "  requires static E.F;" +
-                        "  requires transitive static G.H;" +
+                        "  requires static transitive G.H;" +
```
Comment (matozoid) [inline javaparser-core-testing/src/test/java/com/github/javaparser/modules/ModuleDeclarationTest.java:116]: Ummm, okay...

File: javaparser-core-testing/src/test/java/com/github/javaparser/printer/YamlPrinterTest.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public void testWithColonFollowedBySpaceInValue() {
     public void testWithColonFollowedByLineSeparatorInValue() {
         String expectedOutput = "---" + System.lineSeparator();
         expectedOutput += "root(Type=StringLiteralExpr): " + System.lineSeparator();
-        expectedOutput += "    value: \"a\\\\:\\\\nb\"" + System.lineSeparator();
+        expectedOutput += "    value: \"a\\\\\\\\:\\\\\\\\nb\"" + System.lineSeparator();
```
Comment (matozoid) [inline javaparser-core-testing/src/test/java/com/github/javaparser/printer/YamlPrinterTest.java]: This is from a change by @clusterrun - it seems it made these tests fail and I'm not sure if there are really supposed to be this many escape characters?!

File: javaparser-core-testing/pom.xml [hunk 4]
```diff
@@ -87,6 +87,10 @@
             <groupId>org.junit.jupiter</groupId>
             <artifactId>junit-jupiter-engine</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
+        </dependency>
```
Comment (matozoid) [inline javaparser-core-testing/pom.xml:93]: This was probably caused by my cleanup for the Java 11 compilation PR from a little while ago.

**Comment by ftomassetti** [comment]
I am curious to see the order of magnitude of this. My bad I should have completed the investigation on the missin tests earlier on

**Comment by matozoid** [comment]
> I am curious to see the order of magnitude of this. My bad I should have completed the investigation on the missin tests earlier on

You may blame the maintainer here.

**Comment by ftomassetti** [comment]
We could need some way to make more obvious the number of tests running at each build because this was very, very easy to miss. It seems 3 tests with problems on javaparser-core-testing so that is a relief. javaparser-symbol-solver-testing is going to make the evening interesting :D

**Comment by matozoid** [comment]
Maybe we could have a nice little graph of number of passing unit tests, like on Jenkins :-)

I think JSS is unaffected?

**Comment by ftomassetti** [comment]
If JSS is unaffected I think we should celebrate 💃

**Comment by matozoid** [comment]
I'm not in the mood to celebrate with that stubborn Windows EOL problem :-)

**Comment by ftomassetti** [comment]
All tests passing on Travis, that is a big relief!

**Comment by matozoid** [comment]
Well, I broke a lot more tests on Windows with my last attempt. It's progress!

**Comment by matozoid** [comment]
After torturing my laptop with Intellij Idea in Windows 7 in VMware, I think I have fixed it.

**Comment by ftomassetti** [comment]
Great stuff! Thanks

**Pull request merged**